### PR TITLE
zh-TW: translate memory-management

### PR DESCRIPTION
**Pull request by mgeisler**
Part of #684.

### CODE REVIEW

File: po/zh-TW.po [hunk 1]
```diff
@@ -3529,28 +3563,54 @@ msgid ""
...
 msgid "You allocate and deallocate heap memory yourself."
-msgstr ""
+msgstr "您可以自行分配及釋放堆積記憶體。"
 
 #: src/memory-management/manual.md:5
 msgid ""
 "If not done with care, this can lead to crashes, bugs, security "
 "vulnerabilities, and memory leaks."
-msgstr ""
+msgstr "如果操作時不夠小心，可能會導致當機、錯誤、安全漏洞和記憶體流失。"
```
Comment (johnathan79717) [inline po/zh-TW.po:3605]:
```suggestion
msgstr "如果操作時不夠小心，可能會導致當機、錯誤、安全漏洞和記憶體泄漏。"
```

https://rust-lang.tw/book-tw/ch15-06-reference-cycles.html

File: po/zh-TW.po [hunk 2]
```diff
@@ -3564,22 +3624,33 @@ msgid ""
...
+"}\n"
+"```"
 
 #: src/memory-management/manual.md:21
 msgid ""
 "Memory is leaked if the function returns early between `malloc` and `free`: "
 "the\n"
 "pointer is lost and we cannot deallocate the memory."
 msgstr ""
+"如果函式在 `malloc` 和 `free` 之間提早傳回記憶體，就會發生記憶體流失：指標遺"
```
Comment (johnathan79717) [inline po/zh-TW.po:3643]:
```suggestion
"如果函式在 `malloc` 和 `free` 之間提早傳回記憶體，就會發生記憶體泄漏：指標遺"
```

File: po/zh-TW.po [hunk 3]
```diff
@@ -3564,22 +3624,33 @@ msgid ""
...
+"失，我們也無法釋放記憶體。"
 
 #: src/memory-management/scope-based.md:1
 msgid "# Scope-Based Memory Management"
-msgstr ""
+msgstr "# 範圍式記憶體管理"
```
Comment (johnathan79717) [inline po/zh-TW.po:3648]:
```suggestion
msgstr "# 作用域式記憶體管理"
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -3564,22 +3624,33 @@ msgid ""
...
 
 #: src/memory-management/scope-based.md:3
 msgid ""
 "Constructors and destructors let you hook into the lifetime of an object."
-msgstr ""
+msgstr "建構函式和解構函式可讓您掛入物件的生命週期。"
```
Comment (johnathan79717) [inline po/zh-TW.po:3653]:
```suggestion
msgstr "建構函式和解構函式可讓您掌握物件的生命週期。"
```

File: po/zh-TW.po [hunk 5]
```diff
@@ -3588,17 +3659,19 @@ msgid ""
...
+"只要在物件中包裝指標，即可在物件刪除時釋放記憶體。即使發生例外狀況，編譯器仍"
+"會保證執行這項作業。"
 
 #: src/memory-management/scope-based.md:9
 msgid ""
 "This is often called _resource acquisition is initialization_ (RAII) and "
 "gives\n"
 "you smart pointers."
-msgstr ""
+msgstr "這通常稱為「資源取得即初始化」(RAII)__，且會提供智慧指標。"
```
Comment (johnathan79717) [inline po/zh-TW.po:3670]:
```suggestion
msgstr "這通常稱為「資源取得即初始化」(RAII)，且會提供智慧指標。"
```

Could you tell the translation team to remove future markup ** and __ once they replaced them with 「」?

File: po/zh-TW.po [hunk 6]
```diff
@@ -3529,28 +3563,54 @@ msgid ""
 "    }\n"
 "    ```"
 msgstr ""
+"* 請說明 `String` 是由 `Vec` 支援，因此具有容量和長度，而且還能成長 (前提是可"
+"透過堆積上的重新分配作業進行變動)。\n"
+"\n"
+"* 如有學員問起，您可以說明基礎記憶體是使用[系統分配器]分配的堆積，而自訂分配"
+"器可以使用[分配器 API] 實作。\n"
+"\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:3571]:
```suggestion
"透過堆積上的重新配置作業進行變動)。\n"
"\n"
"* 如有學員問起，您可以說明基礎記憶體是使用[系統配置器]配置的堆積，而自訂配置"
"器可以使用[配置器 API] 實作。\n"
"\n"
```

File: po/zh-TW.po [hunk 7]
```diff
@@ -3529,28 +3563,54 @@ msgid ""
...
 
 #: src/memory-management/manual.md:1
 msgid "# Manual Memory Management"
-msgstr ""
+msgstr "# 手動記憶體管理"
 
 #: src/memory-management/manual.md:3
 msgid "You allocate and deallocate heap memory yourself."
-msgstr ""
+msgstr "您可以自行分配及釋放堆積記憶體。"
```
Comment (johnathan79717) [inline po/zh-TW.po:3599]:
```suggestion
msgstr "您可以自行配置及釋放堆積記憶體。"
```

File: po/zh-TW.po [hunk 8]
```diff
@@ -3529,28 +3563,54 @@ msgid ""
...
 
 #: src/memory-management/manual.md:7
 msgid "## C Example"
-msgstr ""
+msgstr "## C 範例"
 
 #: src/memory-management/manual.md:9
 msgid "You must call `free` on every pointer you allocate with `malloc`:"
-msgstr ""
+msgstr "使用 `malloc` 分配每個指標時，都必須呼叫 `free`："
```
Comment (johnathan79717) [inline po/zh-TW.po:3613]:
```suggestion
msgstr "使用 `malloc` 配置每個指標時，都必須呼叫 `free`："
```

File: po/zh-TW.po [hunk 9]
```diff
@@ -3616,11 +3694,14 @@ msgid ""
 "* At the end of `say_hello`, the `std::unique_ptr` destructor will run.\n"
 "* The destructor frees the `Person` object it points to."
 msgstr ""
+"* `std::unique_ptr` 物件會在堆疊上分配，並指向在堆積上分配的記憶體。\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:3697]:
```suggestion
"* `std::unique_ptr` 物件會在堆疊上配置，並指向在堆積上配置的記憶體。\n"
```

File: po/zh-TW.po [hunk 10]
```diff
@@ -3629,32 +3710,38 @@ msgid ""
...
-msgstr ""
+msgstr "除了手動記憶體管理和範圍式記憶體管理之外，自動記憶體管理是另一種做法："
 
 #: src/memory-management/garbage-collection.md:6
 msgid ""
 "* The programmer never allocates or deallocates memory explicitly.\n"
 "* A garbage collector finds unused memory and deallocates it for the "
 "programmer."
 msgstr ""
+"* 程式設計師一律不會明確分配或釋放記憶體。\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:3735]:
```suggestion
"* 程式設計師一律不會明確配置或釋放記憶體。\n"
```

File: po/zh-TW.po [hunk 11]
```diff
@@ -3683,10 +3775,16 @@ msgid ""
 "* A Rust user can choose the right abstraction for the situation, some even "
 "have no cost at runtime like C."
 msgstr ""
+"* 像 Java 一樣安全又正確，但沒有垃圾收集器。\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:3778]:
```suggestion
"* 像 Java 一樣安全又正確，但沒有垃圾回收機制。\n"
```

File: po/zh-TW.po [hunk 12]
```diff
@@ -3629,32 +3710,38 @@ msgid ""
...
-msgstr ""
+msgstr "# 自動記憶體管理"
 
 #: src/memory-management/garbage-collection.md:3
 msgid ""
 "An alternative to manual and scope-based memory management is automatic "
 "memory\n"
 "management:"
-msgstr ""
+msgstr "除了手動記憶體管理和範圍式記憶體管理之外，自動記憶體管理是另一種做法："
```
Comment (johnathan79717) [inline po/zh-TW.po:3727]:
```suggestion
msgstr "除了手動記憶體管理和作用域式記憶體管理之外，自動記憶體管理是另一種做法："
```

File: po/zh-TW.po [hunk 13]
```diff
@@ -3683,10 +3775,16 @@ msgid ""
 "* A Rust user can choose the right abstraction for the situation, some even "
 "have no cost at runtime like C."
 msgstr ""
+"* 像 Java 一樣安全又正確，但沒有垃圾收集器。\n"
+"* 可以是單一不重複指標、採用參照計數或採用原子參照計數，須視您選擇的抽象方法 "
+"(或抽象方法組合) 而定。\n"
+"* 像 C++ 一樣的範圍式管理，但編譯器會強制遵循完整規定。\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:3781]:
```suggestion
"* 像 C++ 一樣的作用域式管理，但編譯器會強制遵循完整規定。\n"
```

File: po/zh-TW.po [hunk 14]
```diff
@@ -3683,10 +3775,16 @@ msgid ""
...
+"* Rust 使用者可選擇適合情境的抽象方法，部分方法甚至像 C 一樣在執行階段不需付"
+"費。"
 
 #: src/memory-management/rust.md:10
 msgid "It achieves this by modeling _ownership_ explicitly."
-msgstr ""
+msgstr "實現這種記憶體管理的方法是明確建立「擁有權」__模型。"
```
Comment (johnathan79717) [inline po/zh-TW.po:3787]:
```suggestion
msgstr "實現這種記憶體管理的方法是明確建立「所有權」模型。"
```

File: po/zh-TW.po [hunk 15]
```diff
@@ -3616,11 +3694,14 @@ msgid ""
...
 msgstr ""
+"* `std::unique_ptr` 物件會在堆疊上分配，並指向在堆積上分配的記憶體。\n"
+"* `say_hello` 結束時，`std::unique_ptr` 解構函式就會執行。\n"
+"* 解構函式會釋放其指向的 `Person` 物件。"
 
 #: src/memory-management/scope-based.md:25
 msgid ""
 "Special move constructors are used when passing ownership to a function:"
-msgstr ""
+msgstr "將擁有權傳遞至函式時，系統會使用特殊的移動建構函式："
```
Comment (johnathan79717) [inline po/zh-TW.po:3704]:
```suggestion
msgstr "將所有權傳遞至函式時，系統會使用特殊的移動建構函式："
```

**Review by johnathan79717 — CHANGES_REQUESTED** [review]

File: po/zh-TW.po (hunk 5 above)
Comment (mgeisler) [reply po/zh-TW.po:3670]: > Could you tell the translation team to remove future markup ** and __ once they replaced them with 「」?

@henrif75, good advice for next iteration. As I understand it, 「」 is used for emphasis, so the Markdown characters should go afterwards.

File: po/zh-TW.po [hunk 16]
```diff
@@ -3683,10 +3775,16 @@ msgid ""
 "* A Rust user can choose the right abstraction for the situation, some even "
 "have no cost at runtime like C."
 msgstr ""
+"* 像 Java 一樣安全又正確，但沒有垃圾回收機制。\n"
+"* 可以是單一不重複指標、採用參考計數或採用原子性參考計數，須視您選擇的抽象方法 "
```
Comment (johnathan79717) [inline po/zh-TW.po:3779]:
```suggestion
"* 可以是單一不重複指標、採用參考計數或採用原子參考計數，須視您選擇的抽象方法 "
```

**Review by johnathan79717 — APPROVED** [review]

**Pull request merged**